lib/chat_notifier.rb

```diff
@@ -123,10 +159,17 @@ def users_for_channel(exclude:, usernames: nil)
   end
 
   def filter_users_who_can_chat(users)
-    users.select do |user|
+    can_chat = []
+    cannot_chat = []
+    users.each do |user|
       guardian = Guardian.new(user)
-      guardian.can_chat?(user) && guardian.can_see_chat_channel?(@chat_channel)
+      if guardian.can_chat?(user) && guardian.can_see_chat_channel?(@chat_channel)
+        can_chat << user
+      else
+        cannot_chat << user
+      end
     end
+    [can_chat, cannot_chat]
```

Comment:
Just an FYI, there's a neat method just for this:

```ruby
users.partition do |user|
  guardian = Guardian.new(user)
  guardian.can_chat?(user) && guardian.can_see_chat_channel?(@chat_channel)
end
```

Reply:
I love when we get opportunity to use this kind of methods :D

Reply:
Can't believe I didn't know this existed. So freaking awesome.